core/src/main/java/org/elasticsearch/common/lucene/search/function/FieldValueFactorFunction.java

```diff
@@ -116,6 +116,11 @@ protected boolean doEquals(ScoreFunction other) {
                 Objects.equals(this.modifier, fieldValueFactorFunction.modifier);
     }
 
+    @Override
+    protected int doHashCode() {
+        return Objects.hash(boostFactor, field, modifier);
```

Comment:
`Objects.hash` is going to cause an array allocation; if `Object#hashCode` is not going to be called on these objects, that's fine, but if it is and it's going to be called frequently, then we should think about whether or not we should really be using `Objects.hash` here and in the other places in this pull request.